src/FieldWarning/Assets/Units/Component/Movement/Pathfinder.cs

```diff
@@ -55,6 +57,10 @@ public class Pathfinder
         /// </summary>
         private static bool _s_straightStep;
 
+        private Semaphore _PathFinderRunnerSem = new Semaphore(0,1);
```

Comment:
Just use a mutex if the capacity is 1.

Reply:
This not used for multithreading but for signaling. Will use events instead